bpt_visitor.cpp

```diff
@@ -0,0 +1,52 @@
+#include "bpt_visitor.hpp"
+#include "bpt_events.hpp"
+
+namespace bpt {
+
+void visitor::visit(const modload_event& e) {
+    visit(static_cast<const event&>(e));
+}
+
+void visitor::visit(const syscall_event& e) {
```

Comment:
if you just providing a default behavior for the base visitor, then it might be better just to left the implementation empty.